controllers/datafoundation.go

```diff
@@ -171,6 +182,27 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 		}
 
 	} else {
+
+		// Checking for CRDs that are installed after the agent was started and add watchers for the new CRDs
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[StorageClusterCRDName] {
+			crd := apiextensionsv1.CustomResourceDefinition{}
+			crd.Name = StorageClusterCRDName
+			if err := r.get(&crd); err != nil {
+				return ctrl.Result{}, err
+			} else {
+				utils.HandleMissingWatchers(StorageClusterCRDName)
+			}
+		}
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[OCSInitializationCRDName] {
+			crd := apiextensionsv1.CustomResourceDefinition{}
+			crd.Name = OCSInitializationCRDName
+			if err := r.get(&crd); err != nil {
+				return ctrl.Result{}, err
+			} else {
+				utils.HandleMissingWatchers(OCSInitializationCRDName)
+			}
+		}
```

Comment:
@rohan47 
 Requesting again for the 3rd time (3rd review) - Move this code to the start of `reconcilePhases`, This code is a prereq to running any of the offering reconcile logic!